src/wasm/extract.hpp

```diff
@@ -48,13 +48,13 @@ class file_output : private boost::noncopyable {
   boost::uint64_t total_written_;
 
   bool write_;
-  Nonzip &zip_;
-	bool zip_open_;
-	uint32_t zipindex_;
+  ZIPstream *zip_ = NULL;
```

Comment:
Also, in most of the other places the pointer's `*` and reference's `&` stands next to the type rather than the name, so to keep things consistent I'd suggest moving it to the left.
```suggestion
  ZIPstream* zip_ = NULL;
```

Reply:
Ok, let's use nullptr where possible. 
```suggestion
  ZIPstream* zip_ = nullptr;
```